### ISSUE TITLE
after_authenticate_job question

### ISSUE THREAD

**Issue by resistorsoftware** [Shopify/shopify_app#514]
As per the provided documentation:

> AfterAuthenticate Job
> ---------------------

> If your app needs to perform specific actions after it is installed ShopifyApp can queue or run a job of your choosing (note that we already provide support for automatically creating Webhooks and Scripttags). To configure the after authenticate job update your initializer as follows:

Is this job run once and only once when a shop is first installed and has accepted any conditions like charges etc? Or is this job meant to run every time a new session is authenticated? It is confusing terminology to me.

**Comment by kevinhughes27** [comment]
Its ran every time a session is created so it should be idempotent. That why it is called AfterAuthenticate and not AfterInstall but I agree the naming is still not perfect.

**Comment by resistorsoftware** [comment]
Would be nice to have an after_install job so that we can group any "hey, this is a new App, let us bootstrap all the stuff we do once at install time"...

**Comment by kevinhughes27** [comment]
From the session standpoint its pretty hard to detect a fresh install. IMO the best way to do this is an after_create on the Shop model
